### PR TITLE
Make compression middleware prefer brotli over zstd over gzip

### LINKED ISSUE #3187: `.negotiate()` prefers `gzip` over `brotli`
If I enable `actix_web::middleware::Compress` middleware, it's serving `gzip`, even though `brotli` feature is on.

 
## Expected Behavior

I am expecting it to pick the best compression method supported by Browser, and between brotli and gzip, brotli is better (my site compresses to ~196KB with brotli and ~250kb with gzip). 

## Current Behavior

In the 4.4 release and on main, if you look at the test cases, `gzip` is given higher priority over `brotli`.

 https://github.com/actix/actix-web/blob/e50bceb91493087d4059e605d94c13a11b09e747/actix-web/src/http/header/accept_encoding.rs#L378-L385

## Possible Solution

Based on https://news.ycombinator.com/item?id=19678985 I recommend `brotli` > `zstd` > `gzip` as the priority in which one should pick the compression Algo to use if browser supports all three.

Since every browser that supports `brotli` or `zstd` also supports `gzip`, with current implementation of giving `gzip` priority over the two means we are compiling support for `brotli` and `zstd` for no real benefit.

Happy to send a PR if my proposal is accepted. 

## Steps to Reproduce (for bugs)

The test linked above demonstrates the problem. 

## Context

I am trying to optimise a [`fastn`](https://github.com/fastn-stack/fastn/) powered website: [`fastn.com`](https://fastn.com). Currently it is implemented as SSG, deployed on Vercel, and Vercel uses brotli and about 190KB is downloaded for a test page, and when I power it by running `fastn serve` on a Heroku instance, and enable compression middleware, it serves gzipped content, and browser downloads ~250KB of content. 

## Your Environment

- Rust Version (I.e, output of `rustc -V`): 1.73.0
- Actix Web Version: 4.4

Site is deployed on Heroku. [Code in question](https://github.com/fastn-stack/fastn/blob/85f972eb49818423cab055e43dafe71c5af9cc05/fastn-core/src/commands/serve.rs#L737).

### PR DESCRIPTION
**Pull request by amitu**
## PR Type

Bug Fix

## PR Checklist

- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt.
- [x] (Team) Label with affected crates and semver status.

## Overview

Closes #3187.

### CODE REVIEW

File: actix-web/tests/compression.rs [hunk 1]
```diff
@@ -96,7 +96,7 @@ async fn negotiate_encoding_gzip() {
 
     let req = srv
         .post("/static")
-        .insert_header((header::ACCEPT_ENCODING, "gzip,br,zstd"))
+        .insert_header((header::ACCEPT_ENCODING, "gzip"))
```
Comment (amitu) [inline actix-web/tests/compression.rs]: Not too happy with this. Should I add quality specifiers with lesser quality so gzip does get selected despite others specified?

Comment (robjtede) [reply actix-web/tests/compression.rs]: Doing so would make these tests non-representative of real browsers requests.

I'll have a mess about with this PR and update soon.

Comment (robjtede) [reply actix-web/tests/compression.rs]: Okay, I've tweaked the implementation and tests. LMK what you think!

Comment (amitu) [reply actix-web/tests/compression.rs]: Looks good to me.

**Review by robjtede — APPROVED** [review]
thanks 👍🏻 😄

**Pull request merged**